toshi_hazard_store/model/hazard_models_manager.py

```diff
@@ -0,0 +1,281 @@
+"""This module provide classes for managing hazard models as json files.
+
+It is used for local storage of these small and slow moving artefacts.
+
+Each manager class will store a canonical list of instance 'unique_id`s and will
+ provide methods for creating, updating and deleting instances, with
+ serialisation to/from json.
+
+Note:
+ - CLI scripts will use these classes to maintain the metadata for published datasets.
+
+Classes:
+  - CompatibleHazardCalculationManager
+  - HazardCurveProducerConfigManager
+ """
+
+import logging
+from pathlib import Path
+from typing import Any, Dict, List, Union
+
+from pydantic import ValidationError
+
+from toshi_hazard_store.model.hazard_models_pydantic import CompatibleHazardCalculation, HazardCurveProducerConfig
+
+
+class ManagerBase:
+    """Base class for managing storage of configuration models.
+
+    Attributes:
+        storage_folder: The directory where configurations are stored as JSON files.
+    """
+
+    def __init__(self, storage_folder: Path):
+        """Initialize the manager with a specified storage folder."""
+        self.storage_folder = storage_folder
+        if not storage_folder.parent.is_dir():
```

Comment:
should this test instead check if `storage_folder` is a directory if it exists?
```suggestion
        if not storage_folder.exists() and not storage_folder.is_dir():
```